src/psij/executors/flux.py

```diff
@@ -50,175 +57,110 @@ def __init__(self, url: Optional[str] = None,
-        # TODO: why don't we get job.attributes default values here?
-        # TODO: why don't we get job.resources default values here?
-        #
-        if jspec.attributes is None:
-            jspec.attributes = JobAttributes()
-
-        if jspec.resources is None:
-            jspec.resources = ResourceSpecV1(process_count=1,
-                                             processes_per_node=1,
-                                             cpu_cores_per_process=1,
-                                             gpu_cores_per_process=1,
-                                             exclusive_node_use=True)
-        if not jspec.executable:
-            raise InvalidJobException('no executable defined')
-
-        if not jspec.arguments:
-            jspec.arguments = []
-
-        if not isinstance(jspec.resources, ResourceSpecV1):
-            raise InvalidJobException("resource is not ResourceSpecV1")
-        rspec = jspec.resources
-        attr = jspec.attributes
-
-        flux_spec = {
-            "tasks": [{
-                "slot": "task",
-                "count": {
-                    "per_slot": 1
-                },
-                "command": [jspec.executable] + jspec.arguments
-            }],
-            "attributes": {
-                "system": {
-                    # FIXME: flux hangs on `None` values
-                    # "cwd": jspec.directory,
-                    "duration": attr.duration.total_seconds()
-                }
-            },
-            "version": 1,
-            "resources": [{
-                "count": rspec.process_count,
-                "type": "slot",
-                "label": "task",
-                "with": [{
-                    "count": rspec.cpu_cores_per_process,
-                    "type": "core"
-                }, {
-                    "count": rspec.gpu_cores_per_process,
-                    "type": "gpu"
-                }]
-            }]
-        }
-
-        return flux_spec
+        assert job.spec
+        assert job.spec.attributes
+        if isinstance(job.spec.resources, ResourceSpecV1):
+            resources = job.spec.resources
+        elif isinstance(job.spec.resources, ResourceSpec):
+            raise InvalidJobException(
+                f"ResourceSpec version {job.spec.resources.version} not supported"
+            )
+        else:
+            resources = ResourceSpecV1(process_count=1, cpu_cores_per_process=1)
+        if resources.exclusive_node_use:
+            warnings.warn(
+                "Flux does not support exclusive_node_use=True, ignoring...",
+                UserWarning,
+            )
+        if resources.processes_per_node:
+            raise InvalidJobException("Flux does not support processes_per_node")
+        if not job.spec.executable:
+            raise InvalidJobException("Job must have an executable")
+        argv = list(job.spec.arguments) if job.spec.arguments else []
+        argv.insert(0, job.spec.executable)
```

Comment:
`[job.spec.executable] + job.spec.arguments` (or similar) will save you one O(n) operation.

Reply:
That's true, will fix, my bad assumption about Python list performance. Verified with python wiki.

Reply:
Won't make much of a difference, since you already have an O(n) and n is small, so feel free to keep, but I thought the alternative would also look nicer.

Reply:
Yeah, I am a bit to lazy to rebase right now, so I will just merge. If I remember I will make the change later.